CT-1391 log new sar case

misaka opened the pull request:
Another attempt at getting these validations right. Further comments in the commits.

aliuk2012 commented:
@misaka @EdwardAndress  I think we should review and merge this and then tweak the rest of assign page later.

aliuk2012 commented:
I've split this ticket into two separate JIRA tickets and so the understanding work thats blocking this PR/work has been moved into ct-353.

misaka commented on app/views/assignments/new.html.slim:
```diff
@@ -1,3 +1,7 @@
-	= f.collection_select :assignee_id, User.drafters, :id, :email
-	= f.submit class: 'button'
+
+  -# f.radio_button_fieldset :assignee_id, choices: User.drafters.tap(&:email).to_sym
```
Do we want this commented-out line?

aliuk2012 replied:
Fixed it

misaka committed:
CT-1391 add more validations for name, email, third_party, etc.

misaka committed:
CT-1391 make third_party and reply_method validations better

Further clean-up and improvements to validations and how we present them. Here
we add the gem 'humanize_boolean' to tidy up how we present 'third_party' values
and allow 'reply_method' to be presented as 'send_by_email' and 'send_by_post'
instead of 'email' and 'post'. By doing this, we remove the translations
third_party and reply_method in en.helpers.label.case_sar, which were blocking
presenting the validation errors correctly: when the values for these
translations are a hash, as was the case, the name of the attribute in the
validation error text is stripped out by the UK Gov FormBuilder (see PR 67
https://github.com/ministryofjustice/govuk_elements_form_builder/pull/67).

misaka committed:
CT-1391 move JSON attributes into Case::FOI::Standard and Case::SAR

The correct way to use these JSON attributes is to not have them show up on
every case, by putting them in Case::Base, but to have them specific to the
different sub-classes of cases. This does mean that common attributes have to be
redefined in multiple places, but this is probably ok as attributes common to
all cases should probably be defined as columns anyway.

misaka committed:
CT-1391 minor fixes for SAR cases

- rename case_foi model errors in translations back to case, this was a
  miss-understanding of what would be affected by this change
- remove validations for SAR attributes in FOI model specs, these attributes
  cannot be set now that SAR JSON attributes are only defined in the Case::SAR
  model
- third_party now has the value true/false

misaka committed:
CT-1391 remove SAR attributes from admin's case creation

misaka committed:
CT-1391 add specs for SAR attribute validations
